lib/grammers-client/src/types/inline_query.rs

```diff
@@ -0,0 +1,169 @@
+use crate::{client::Client, utils::generate_random_id, InputMessage};
+use grammers_mtsender::InvocationError;
+use grammers_tl_types as tl;
+
+/// Represents an inline query update, which occurs when you sign in as a bot and a user sends an
+/// inline query such as `@bot query`.
+pub struct InlineQuery {
+    query: tl::types::UpdateBotInlineQuery,
+    client: Client,
+}
+
+/// An inline query answer builder.
+pub struct Answer {
+    request: tl::functions::messages::SetInlineBotResults,
+    client: Client,
+}
+
+impl InlineQuery {
+    pub(crate) fn new(client: &Client, query: tl::types::UpdateBotInlineQuery) -> Self {
+        Self {
+            query,
+            client: client.clone(),
+        }
+    }
+
+    // User that sent the query.
+    pub fn user_id(&self) -> i32 {
+        self.query.user_id
+    }
+
+    // The text of the inline query.
+    pub fn text(&self) -> &str {
+        self.query.query.as_str()
+    }
+
+    /// Answer the inline query.
+    // TODO: add example
+    pub fn answer(&self, results: Vec<tl::enums::InputBotInlineResult>) -> Answer {
```

Comment:
I'd prefer having a `struct InlineResult(tl::enums::InputBotInlineResult)`, and then the custom types like `Article` implement into `InlineResult`. I would like to avoid having any raw type (from `tl::`) in the public API at all.

Reply:
Done.